doc/rst/developer/bestPractices/ContributorInfo.rst

```diff
@@ -9,184 +9,274 @@ The git repository for the project is hosted on Github at
 ------------------
 
 Overview:
 
-#. `Fork repo on github`_
-#. `Create new branch`_
-#. `Submit pull request`_
-#. `Test changes`_ (can be done by team at Cray)
-#. `Mail chapel-developers@ for review and merge`_
+#. `Discuss design`_ changes or big development efforts
+#. `Set up a branch for development`_
 
-.. _Fork repo on github:
+   #. `Getting set up`_
+   #. `Create new branch`_
 
-Fork repo on github
-~~~~~~~~~~~~~~~~~~~
+#. `Develop and test contributions locally`_
 
-These are expected to evolve over time as the governance of Chapel is migrated
-from Cray to an external/community body (the major elements are likely to be
-similar, though the specific people involved are likely to change and grow).
+   #. `Adding new tests`_
 
-* `Set up a github account`_. The "Free" plan is sufficient for contributing to
-  Chapel.
+#. `Request feedback on your changes`_
 
-* Make sure you have configured your environment to work with git. See
-  :ref:`Initial Git Set up <initial_git_setup>` instructions.
+   #. `Push your work`_
+   #. `Ask for feedback early`_ (optional)
+   #. `Submit pull request`_
+   #. `Find a reviewer`_
+   #. `Work with your reviewers`_
 
-* If you will need commit/push access to the main repository,
-  `chapel-lang/chapel`_, send a request including your github username to
-  chapel_admin _at_ cray.com.
+#. `Test changes`_
+#. `Getting ready to merge`_
+#. `Watch automatic testing`_ and address issues
 
-  .. note::
+`HOWTO/Git/GitHub details`_
 
-    You do not need commit/push access to the main repo in order to contribute
-    code. Reviewers on the core team can pull, review, and merge your pull
-    requests.
+`Policy details`_
 
-* If you're working on a long-term effort, announce it on the
-  chapel-developers_ to make sure toes are not being stepped on, work is not
-  being pursued redundantly, etc.  Similarly, fundamental changes to the
-  language or architecture should be circulated to the chapel-developers_
-  and/or chapel-users_ lists to make sure effort is not wasted.
+.. _Discuss design:
 
-* If your work will require committing any third-party code that you are not
-  developing yourself (or code that you've developed as a standalone package),
-  alert the chapel-developers_ mailing list of this as, presently, such code
-  packages must be approved by Cray leadership before being committed.
+Discuss design changes or big development efforts
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-* Sign a Chapel contributor's agreement and mail it, with your GitHub
-  ID, using the instructions here:
+Before embarking on a big project, please open an issue on the git repository
+and/or send an email to chapel-developers_ with a prefix of ``[Design]`` in the
+subject header for discussion with the community and to ensure that you are
+aware of any parallel efforts in that area.  If the change will impact the user
+experience, it would be appropriate to additionally send this email to
+chapel-users_.
```

Comment:
I think I'd word this less strongly - something like "It may also be good idea to ask the user community (on chapel-users_, say) what they think about a project or language direction".
I don't think we want to suggest that *all* language changes (even) should have emails to chapel-users.